x/interchainstaking/keeper/keeper.go

```diff
@@ -347,6 +353,14 @@ func (k *Keeper) SetValidatorForZone(ctx sdk.Context, zone *types.Zone, data []b
 		}
 	}
 
+	// checking whether validator was tombstoned or not
+	if val.Jailed {
```

Comment:
This wants to happen in the conditional at line 298 (`if !val.Jailed && validator.Jailed`) in order to avoid repeated lookups every time the validator VP changes once jailed (in this case, `val.Jailed == true`, but ` !val.Jailed && validator.Jailed` will evaluate to `true` only on the first run.

Reply:
We will also need to call this in l275, if the _new_ validator is in a jailed state.